tensorflow/lite/micro/kernels/softmax.cc

```diff
@@ -73,6 +101,71 @@ TfLiteStatus CalculateSoftmaxParams(TfLiteContext* context,
 
 }  // namespace
 
+void* SoftmaxInit(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  void* raw_op_data =
+      context->AllocatePersistentBuffer(context, sizeof(SoftmaxOpData));
+  if (raw_op_data == nullptr) {
+    return nullptr;
+  }
+  SoftmaxOpData* op_data = static_cast<SoftmaxOpData*>(raw_op_data);
+  const TfLiteType input_type = context->tensors[0].type;
```

Comment:
Hi @petewarden. The micro example tests segfaulted on this line. It turns out that micro interpreter calls the registration.init(...) as part of its AllocateTensors method (https://github.com/tensorflow/tensorflow/blob/8c6f2d55762f3fc94f98fdd8b3c5d59ee1276dba/tensorflow/lite/micro/micro_interpreter.cc#L275). However the interpreter doesn't seem to populate the context with input and output tensors at all (instead it exposes them as inputs(index) and outputs(index) methods only. Could you please advise how this can be resolved? We can either take the easy route that allocates LUTs regardless types (memory waste), or change the interpreter so that it populates the context with input+output tensors so that we can obtain the type info within init() here, although I don't know the implications of the latter approach.

Reply:
@njeffrie it looks like this is related to the EvalTensors changes?

Reply:
Mark this thread as resolved as the issue continues in the discussion below